src/main/kotlin/org/openmbee/flexo/sysmlv2/apis/CommitApi.kt

```diff
@@ -346,56 +391,79 @@ fun Route.CommitApi() {
 
                 // generate the RDF data for SPARQL INSERT clause
                 inserts.add("""
-                    ${elementNode.stringify()} ${joinToString(" ;\n\t") { pair ->
+                    ${elementNode.stringify()} ${joinToString(" ;\n") { pair ->
                         pair.first.stringify()+" "+pair.second.joinToString(", ") { it.stringify() }
-                    }} .
+                    }.reindent(6)} .
                 """.trimIndent())
             }
         }
-/*    //no need to do any locks since we're always commiting to head of branch
-        // first, lock the given commit
-        val flexoResponseLock = flexoRequestPost {
-            orgPath("/repos/$projectId/locks")
 
-            turtle {
-                """
-                    <> mms:commit mor-commit:${commit.previousCommit} .
-                """.trimIndent()
+        // build SPARQL UPDATE string
+        var sparqlUpdateString = """
+            ${DEFAULT_PREFIX_MAPPING.nsPrefixMap.filter { (id, iri) ->
+                iri.startsWith(SYSMLV2.BASE)
+            }.toList().joinToString("\n") { (id, iri) ->
+                "prefix $id: <$iri>"
+            }.reindent(3)}
+
+            delete {
+                ?element_n ?element_p ?element_o .
+
+                ?incoming ?incoming_p ?element_del ;
+                    ?incoming_order_p ?incoming_order_o .
             }
-        }
+            insert {
+                ${inserts.joinToString("\n\n").reindent(4)}
+            }
+            where {
+                optional {
+                    ?element_n ?element_p ?element_o .
+                }
+                
+                values ?element_n {
+                    ${values.joinToString("\n").reindent(5)}
+                }
+        """
 
-        // forward failures to client
-        if(flexoResponseLock.isFailure()) {
-            return@post forward(flexoResponseLock)
-        }
+        // only delete incoming if there are elements being deleted
+        if(deleteIncoming.isNotEmpty()) {
+            sparqlUpdateString += """
+                optional {
+                    ?incoming ?incoming_p ?element_del ;
+                        ?incoming_order_p ?incoming_order_o .
```

Comment:
still confused by this - `?incoming_relation_p` in the filter should be `?incoming_p` i think, but even then, if the annotation is an array with more than 1 item, this would remove the entire annotation and not just the item for `?element_del`